ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -1257,35 +1257,35 @@ func performChangeAndReboot(ctx context.Context, client *st.IBMPIInstanceClient,
 
 }
 
-func isWaitforPIInstanceUpdate(ctx context.Context, client *st.IBMPIInstanceClient, id string) (interface{}, error) {
+func isWaitforPIInstanceUpdate(ctx context.Context, client *instance.IBMPIInstanceClient, id string, timeout time.Duration) (interface{}, error) {
 	log.Printf("Waiting for PIInstance (%s) to be ACTIVE or SHUTOFF AFTER THE RESIZE Due to DLPAR Operation ", id)
 
 	stateConf := &retry.StateChangeConf{
-		Pending:    []string{"RESIZE", "VERIFY_RESIZE"},
-		Target:     []string{"ACTIVE", "SHUTOFF", helpers.PIInstanceHealthOk},
+		Pending:    []string{State_Resize, State_VerifyResize},
+		Target:     []string{State_Active, State_Shutoff, OK},
 		Refresh:    isPIInstanceShutAfterResourceChange(client, id),
-		Delay:      10 * time.Second,
+		Delay:      Timeout_Delay,
```

Comment:
Any reason to change from 10sec to 60sec? Delete and Update are quick process. We might need big delays in the creating process as it takes around 30min to provision.

Reply:
@michaelkad

Reply:
@yussufsh The majority of these processes operate asynchronously. Checking every 10-30 seconds often fails to yield a prompt response, hence the decision to extend the interval to 60 seconds—a more effective compromise. This interval represents the delay between each polling cycle as we await the desired state, whether it's active, deleted, or any other specified condition.

Reply:
And rather than hardcoding the `Timeout` we are using the value we have defined or what the user is passing in.
Using ` timeout time.Duration` which then get called with the respective timeout `_, err = isWaitforPIInstanceUpdate(ctx, client, id, d.Timeout(schema.TimeoutUpdate))`